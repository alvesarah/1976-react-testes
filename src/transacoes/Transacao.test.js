import React from "react";
import { render } from "@testing-library/react";
import Transacoes from "./Transacoes";

describe("Componente Lista de Transações do extrato", () => {
  it("O snapshot deve sempre permanecer o mesmo", () => {
    const transacoes = [
      {
        id: 1,
        transacao: "saque",
        tipo: "saque",
        valor: "10",
        data: "01/01/2020",
      },
      {
        id: 2,
        transacao: "deposito",
        tipo: "deposito",
        valor: "10",
        data: "01/01/2020",
      },
    ];
    const { container } = render(<Transacoes transacoes={transacoes} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
