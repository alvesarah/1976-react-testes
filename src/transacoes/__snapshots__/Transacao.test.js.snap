// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Lista de Transações do extrato O snapshot deve sempre permanecer o mesmo 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      01/01/2020
    </p>
    <p>
      saque
    </p>
    <p>
      R$ 
      10
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      01/01/2020
    </p>
    <p>
      deposito
    </p>
    <p>
      R$ 
      10
    </p>
  </div>
</div>
`;
